File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        seens = {}

        for i in range(len(nums)):
            seens[nums[i]] = seens.get(nums[i],0)+1

        for i in range(len(nums)):
            if(seens[nums[i]] > len(nums)/2):
                return nums[i]
        
        return -1

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> seens = new HashMap<>();

        for (int i = 0; i < nums.length;i++)
        {
            seens.put(nums[i],seens.getOrDefault(nums[i],0)+1);
        }
        for (int i = 0; i < nums.length; i++)
        {
            if(seens.get(nums[i])  > nums.length /2)
                    return nums[i];
        }
        return -1;
    }
}